physics/smoke_dust/module_add_emiss_burn.F90

```diff
@@ -6,217 +6,152 @@ module module_add_emiss_burn
-    real(kind_phys), PARAMETER :: fact_numc= 1.e-9*6.0/pic*exp(4.5*log(sgem_c)**2)/dgvem_c**3       ! coarse mode
-
-    real(kind_phys), PARAMETER :: dens_oc_aer=1.4e3, dens_ec_aer=1.7e3  ! kg/m3
-!    real(kind_phys), PARAMETER :: rinti=2.1813936e-8, cx=2.184936* 3600, timeq_max=3600.*24.   ! constants for the diurnal cycle calculations
-    real(kind_phys), PARAMETER :: ax1=531., cx1=7800.  ! For cropland, urban and small fires
-!    real(kind_phys), PARAMETER :: rinti=2.1813936e-8, ax2=3200., const2=100., coef2=10.6712963e-4, cx=2.184936* 3600, timeq_max=3600.*24.
-    real(kind_phys), PARAMETER :: rinti=2.1813936e-8, ax2=3400., const2=130., coef2=10.6712963e-4, cx2=7200., timeq_max=3600.*24.    ! New parameters
-    real(kind_phys), PARAMETER :: sc_me= 4.0, ab_me=0.5     ! m2/g, scattering and absorption efficiency for smoke
-
-!   Parameters used for the wfa and ifa in mp physics per Trude E. (NCAR)
-!   Water friendly: radius: 0.04 micron, standard deviation: 1.8, kappa (for hygroscopic growth): 0.2, real index of refraction: 1.53, imaginary index of refraction: 1e-7
-!   Ice friendly: radius: 0.4 micron, standard deviation: 1.8, kappa : 0.04, real index of refraction: 1.56, imaginary index of refraction: 3e-3
+         INTENT(INOUT ) ::                                   ebu
 
+   real(kind_phys), DIMENSION(ims:ime,jms:jme), INTENT(IN) ::  xlat,xlong, swdown 
+   real(kind_phys), DIMENSION(ims:ime,jms:jme), INTENT(IN) :: hwp, peak_hr, fire_end_hr, ebu_in !, vfrac
+   real(kind_phys), DIMENSION(ims:ime,jms:jme), INTENT(INOUT)    ::  coef_bb_dc    ! RAR:
+   real(kind_phys), DIMENSION(ims:ime,jms:jme), INTENT(IN) :: hwp_prevd
+ 
+   real(kind_phys), DIMENSION(ims:ime,kms:kme,jms:jme), INTENT(IN) :: dz8w,rho_phy  !,rel_hum
+   real(kind_phys), INTENT(IN) ::  dtstep, gmt
+   real(kind_phys), INTENT(IN) ::  time_int,pi       ! RAR: time in seconds since start of simulation
+   INTEGER, DIMENSION(ims:ime,jms:jme), INTENT(IN) :: fire_type
+   integer, INTENT(IN) ::  ebb_dcycle     ! RAR: this is going to be namelist dependent, ebb_dcycle=means 
+   real(kind_phys), DIMENSION(ims:ime,jms:jme), INTENT(INOUT) :: fhist
+!>--local 
+   integer :: i,j,k,n,m
+   real(kind_phys) :: conv_rho, conv, dm_smoke, dc_hwp, dc_gp, dc_fn !daero_num_wfa, daero_num_ifa !, lu_sum1_5, lu_sum12_14
+   
+   INTEGER, PARAMETER :: kfire_max=51    ! max vertical level for BB plume rise
+    
     !    real, parameter :: cx        =  2.184936 * 3600., rinti     =  2.1813936e-8    , ax        =  2000.6038
     ! bx_bburn  =  20.041288 * 3600.,  RAR: this depends on the vegetation class, location (local time) etc.
-    real(kind_phys) :: timeq, dt1,dt2,dtm         ! For BB emis. diurnal cycle calculation
+    real(kind_phys) :: timeq, fire_age, age_hr,  dt1,dt2,dtm         ! For BB emis. diurnal cycle calculation
+   ! For Gaussian diurnal cycle
+   REAL, PARAMETER :: sc_factor=1.  ! to scale up the wildfire emissions, TBD later
```

Comment:
Please add the precision to these declarations.

Reply:
@dustinswales Yes, I will add precision to these declarations.